src/content/blog/2024-12-04-differences-between-linting-and-type-checking.md

```diff
@@ -26,19 +26,19 @@ There are three forms of static analysis commonly used in web development projec
 We'll focus in this blog post on *linters*, namely ESLint, and *type checkers*, namely [TypeScript](https://typescriptlang.org).
 ESLint and TypeScript are often used together to detect defects in code.
 
-## Differences in Purpose
+## Digging deeper into linting vs. type checking
 
 Type checkers make sure the *intent* behind values in code matches the *usage* of those values.
-They check that code is "type-safe": meaning values are used according to the ways their types describe as being allowed.
+They check that code is "type-safe": values are used according to how their types are described as being allowed.
 
 Type checkers do not attempt to look for defects that are only likely, rather than certain.
 Nor do type checkers enforce subjective opinions that might change between projects.
 That means TypeScript does not attempt to enforce general best practices — only that values are used in type-safe ways.
```

Comment:
This is where I think it might be helpful to mention the different types of static analysis TypeScript does. `tsc` actually has added in a bunch of linting rules (tracking unused variables, conditions that always evaluate to true, etc.), so it does try to enforce some best practices too.

Otherwise, you may want to be more specific, something like "The type checking in TypeScript..."

Reply:
👍 I don't want to weigh down the blog post with too deep a dive into the nuances of TypeScript. "The type checking in TypeScript..." resonates well with me.